toolchain/check/subst.h

```diff
@@ -32,16 +32,26 @@ class SubstInstCallbacks {
     // Attempt to substitute again on the resulting instruction, acting like
     // recursion on the instruction itself.
     SubstAgain,
+    // Attempt to substitute into the operands of the instruction. If the
+    // operands change, then the rebuilt instruction will be given to `Subst`
+    // again afterward. This allows for the uncommon case of substituting from
+    // the inside out.
```

Comment:
If I'm understanding correctly, the condition here isn't quite "If the operands change", but instead it's the subtler "If `Rebuild` or `ReuseUnchanged` returns a different `InstId` than the original instruction ID they were given", right?